test/functional/fetchers_test.rb

```diff
@@ -54,4 +54,52 @@ def assert_fetcher_failed_cleanly(run_result, error_regex, profile_location)
     end
 
   end
+
+  # Refs #4727
+  describe "when a archive is available of an unfetchable profile with --airgap" do
+
+    let(:invocation) { "archive #{fetcher_profiles}/#{profile_name} --airgap" }
+
+    def teardown
+      FileUtils.rm_rf "#{fetcher_profiles}/#{profile_name}/vendor"
+      FileUtils.rm_f "#{profile_name}-0.1.0.tar.gz"
```

Comment:
*scary* we need to be really careful with rm_rf and rm_f especially since we're not using temp directories anywhere here (we should though).

Reply:
Agreed